internal/services/storage/storage_account_resource.go

```diff
@@ -1246,6 +1247,20 @@ func resourceStorageAccount() *pluginsdk.Resource {
 						return fmt.Errorf("`large_file_share_enabled` cannot be disabled once it's been enabled")
 					}
 				}
+
+				if d.Get("access_tier") != "" {
+					if accountKind := d.Get("account_kind").(string); !slices.Contains(
+						[]storage.Kind{
+							storage.KindBlobStorage,
+							storage.KindStorageV2,
+							storage.KindFileStorage,
+						},
```

Comment:
this same slice of values is defined 3 times - can we pull this out into a constant/variable/function so that these don't fall out of sync?

Reply:
@tombuildsstuff Though the storage kinds that support access_tier (2 places) and hns (1 place) are the same, while they are for different checks, I've abstract them to two variables of list, though same value.